samples/Compute/allowed-image-publishers/azurepolicy.json

```diff
@@ -0,0 +1,43 @@
+{
+    "type": "Microsoft.Authorization/policyDefinitions",
+    "name": "allowed-image-publishers-policy",
+    "properties": {
+        "displayName": "Only allow a certain image publishers offerings to be deployed",
+        "policyType": "Custom",
+        "description": "This policy ensures that only allowed image publisher offerings are selected from the image repository",
+        "Mode": "All",
+        "parameters": {
+            "listOfAllowedimagePublisher": {
+                "type": "Array",
+                "metadata": {
+                    "description": "The list of publishers to audit against.",
+                    "strongType": "imagePublisher",
+                    "displayName": "Allowed image publisher"
+                }
+            }
+        },
+        "policyRule": {
+            "if": {
+                "allOf": [{
+                        "field": "type",
+                        "in": [
+                            "Microsoft.Compute/virtualMachines",
+                            "Microsoft.Compute/VirtualMachineScaleSets"
+                        ]
+                    },
+                    {
+                        "not": {
+                            "allOf": [{
```

Comment:
this allOf isn't necessary as it only contains one clause.  It doesn't affect the functionality of the policy but it might be confusing to readers